numpy/_core/src/multiarray/stringdtype/casts.cpp

```diff
@@ -1887,29 +2160,36 @@ get_cast_spec(const char *name, NPY_CASTING casting,
     return ret;
 }
 
-PyArray_DTypeMeta **
-get_dtypes(PyArray_DTypeMeta *dt1, PyArray_DTypeMeta *dt2)
-{
-    PyArray_DTypeMeta **ret = PyMem_Malloc(2 * sizeof(PyArray_DTypeMeta *));
-
-    ret[0] = dt1;
-    ret[1] = dt2;
+// Check if the argument is inf.
+// Needed to ensure the return type is an int, which is needed
+// for getStringToFloatCastSpec.
+template<typename T>
+int is_inf(T x) {
+    return std::isinf(x);
+}
 
-    return ret;
+// Cast the argument to the given type.
+// Needed because getStringToFloatCastSpec takes a function rather than
+// a type (for casting) as its double_to_float template parameter
+template<typename TNpyType>
+TNpyType to_float(double x) {
+    return static_cast<TNpyType>(x);
 }
 
 PyArrayMethod_Spec **
```

Comment:
```suggestion
NPY_NO_EXPORT PyArrayMethod_Spec **
```

Oops, I guess this was wrong before